### DISCUSSION TITLE
Kedro Newb Struggling To Get Catalog in Spaceflights tutorial working

### DISCUSSION THREAD

**Discussion by dwinski** [kedro-org/kedro#1116] [Idea]
I'm new to kedro and running 0.17.6 on a Windows machine inside a conda environment with Python 3.8.12 where I have pip installed all the requirements from the requirements.txt file that was created with "kedro new".  I also tried to install the pandas.CSVDataSet and pandas.ExcelDataSet classes by modifying the line in the requirements.txt file per the spaceflights tutorial to:
kedro[pandas.csvdataset,pandas.exceldataset]==0.17.6

I'm trying to follow the spaceflights tutorial but catalog in the kedro ipython session is not working (NameError: name 'catalog' is not defined) even when I try the magic commands below per this stackoverflow thread:  https://stackoverflow.com/questions/70080915/kedro-context-and-catalog-missing-from-ipython-session
```
%load_ext kedro.extras.extensions.ipython
%reload_kedro <path_to_project_root>

```
However, when I run:
`%reload_kedro <path_to_project_root>`

I get an exception: 
DataSetError: An exception occurred when parsing config for DataSet 'companies': Class 'pandas.CSVDataSet' not found or one of its dependencieshas not been installed.

Would really like to get through this tutorial.  Any help would be appreciated.

**Comment by datajoely** [comment]
Hi @dwinski!

Did you run `kedro install` after modifying the requirements.txt? You can also skip this part and quickly run `pip install "kedro[pandas.CSVDataSet, pandas.ExcelDataSet]"==0.17.6` 

That catalog is not defined error is because Kedro cannot create the catalog without the right requirements installed

**Reply by dwinski** [reply]
Thanks for your suggestion @datajoely but I did run the install after modifying the line.  I also just tried it again from terminal in conda environment for tutorial:
`pip install kedro[pandas.csvdataset,pandasexceldataset]==0.17.6 `
and all the requirements are already satisfied.  I did get one warning as below:
WARNING: kedro 0.17.6 does not provide the extra 'pandasexceldataset'

I am also still seeing the same catalog error and DataSetError as above when I try the kedro ipython commands in the tutorial and the magic commands.

**Reply by datajoely** [reply]
You're missing a `.` in `pandas.exceldataset`

**Reply by datajoely** [reply]
This is helpful feedback though - I'm currently in the process of updating these docs as we speak #1112

**Reply by dwinski** [reply]
Doh, good catch.  I just added it back and did the pip install again but but still the same errors about 'catalog' is not defined and DataSetError:...Class 'pandas.CSVDataSet' not found or one of its dependencies has not been installed.

**Reply by datajoely** [reply]
Can you run `pip freeze` and paste the results?

**Reply by datajoely** [reply]
I'm being silly the capitalisation is important - you need to do this:

`pip install "kedro[pandas.CSVDataSet, pandas.ExcelDataSet]"==0.17.6`

**Reply by dwinski** [reply]
Yes, will post in a few minutes

**Reply by dwinski** [reply]
Ok, believe I originally tried with capitalization but going to try again.

**Reply by dwinski** [reply]
Same issues after capitalizing.  To be clear, I did following:
`pip install "kedro[pandas.CSVDataSet,pandas.ExcelDataSet]"==0.17.6`

The space after the comma breaks the command so I had to leave it out.

**Reply by dwinski** [reply]
Here's pip freeze:
anyconfig==0.10.1
anyio==3.4.0
appdirs==1.4.4
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.1
atomicwrites==1.4.0
attrs==21.2.0
Babel==2.9.1
backcall==0.2.0
binaryornot==0.4.4
black==21.5b1
bleach==4.1.0
cachetools==4.2.4
certifi==2021.10.8
cffi==1.15.0
chardet==4.0.0
charset-normalizer==2.0.9
click==7.1.2
colorama==0.4.4
cookiecutter==1.7.3
coverage==6.2
debugpy==1.5.1
decorator==5.1.0
defusedxml==0.7.1
dynaconf==3.1.7
entrypoints==0.3
flake8==3.9.2
fsspec==2021.11.1
gitdb==4.0.9
GitPython==3.1.24
idna==2.10
importlib-resources==5.4.0
iniconfig==1.1.1
ipykernel==6.6.0
ipython==7.30.1
ipython-genutils==0.2.0
ipywidgets==7.6.5
isort==5.10.1
jedi==0.18.1
Jinja2==3.0.3
jinja2-time==0.2.0
jmespath==0.10.0
json5==0.9.6
jsonschema==4.2.1
jupyter==1.0.0
jupyter-client==6.1.12
jupyter-console==6.4.0
jupyter-core==4.9.1
jupyter-server==1.13.1
jupyterlab==3.2.5
jupyterlab-pygments==0.1.2
jupyterlab-server==2.9.0
jupyterlab-widgets==1.0.2
kedro==0.17.6
kedro-telemetry==0.1.3
MarkupSafe==2.0.1
matplotlib-inline==0.1.3
mccabe==0.6.1
mistune==0.8.4
mypy-extensions==0.4.3
nbclassic==0.3.4
nbclient==0.5.9
nbconvert==6.3.0
nbformat==5.1.3
nbstripout==0.5.0
nest-asyncio==1.5.4
notebook==6.4.6
numpy==1.21.4
packaging==21.3
pandas==1.3.5
pandocfilters==1.5.0
parso==0.8.3
pathspec==0.9.0
pep517==0.12.0
pickleshare==0.7.5
pip-tools==6.4.0
pluggy==0.13.1
poyo==0.5.0
prometheus-client==0.12.0
prompt-toolkit==3.0.24
py==1.11.0
pycodestyle==2.7.0
pycparser==2.21
pyflakes==2.3.1
Pygments==2.10.0
pyparsing==3.0.6
pyrsistent==0.18.0
pytest==6.2.5
pytest-cov==3.0.0
pytest-mock==1.13.0
python-dateutil==2.8.2
python-json-logger==2.0.2
python-slugify==5.0.2
pytz==2021.3
pywin32==302
pywinpty==1.1.6
PyYAML==5.4.1
pyzmq==22.3.0
qtconsole==5.2.2
QtPy==1.11.3
regex==2021.11.10
requests==2.25.1
rope==0.21.1
Send2Trash==1.8.0
six==1.16.0
smmap==5.0.0
sniffio==1.2.0
terminado==0.12.1
testpath==0.5.0
text-unidecode==1.3
toml==0.10.2
tomli==2.0.0
toposort==1.7
tornado==6.1
traitlets==5.1.1
typing_extensions==4.0.1
urllib3==1.26.7
wcwidth==0.2.5
webencodings==0.5.1
websocket-client==1.2.3
widgetsnbextension==3.5.2
wincertstore==0.2
xlrd==1.2.0
XlsxWriter==1.4.5
zipp==3.6.0

**Reply by datajoely** [reply]
So I've run the following commands in sequence:

1. `conda create -n spaceflights-dec15 python=3.8 -y`
2. `conda activate spaceflights-dec15`
3. `pip install "kedro[pandas.CSVDataSet,pandas.ExcelDataSet]"==0.17.6`
4. `pip freeze | grep -Eo "(kedro|pandas|xlrd).+"`

This shows with a new environment Kedro has installed the correct dependencies:
```bash
kedro==0.17.6
pandas==1.3.5
xlrd==1.2.0
```

**Reply by datajoely** [reply]
Okay that's good! It means the dependencies are installed we can see pandas, xlrd and XlsxWriter.

Let's work out why `kedro ipython` isn't working

**Reply by dwinski** [reply]
Ok, glad dependencies are correct.  Are you able to reproduce the catalog error in kedro ipython in your spaceflights-dec15 environment?

**Reply by dwinski** [reply]
Btw, I was doing the spaceflights tutorial from scratch (did not use starter with "kedro new")

**Reply by datajoely** [reply]
Hi @dwinski I'm not exactly sure what's gone wrong here - but I've run through the tutorial until the point you've got stuck and pushed my progress to this repository:

https://github.com/datajoely/spaceflights-dwinski/

At this point I'm able to do the `kedro ipython` workflow. We've confirmed that you have the right dependencies installed, does your catalog look very different to mine? Is the data in the same folder?

![image](https://user-images.githubusercontent.com/35801847/146270923-7890112c-7497-45da-9f32-17e90ed9b361.png)

**Reply by dwinski** [reply]
Thanks for trying, good to know it's working for you.  My catalog looks exactly the same as yours but still getting the same two kedro ipython errors.  After you did kedro new did you also do (this is what I did): 
`pip install -r src/requirements.txt   `
Could something else in the requirements.txt file have broken the environment?

**Reply by dwinski** [reply]
I'm going to try to recreate environment from scratch per your message above.

**Comment by dwinski** [comment]
I'm trying to recreate your steps from to create the spaceflights-dec15 environment but if I only do:
`pip install kedro[pandas.csvdataset,pandasexceldataset]==0.17.6`

then it appears I won't get IPython installed.  So appears I do need to also do:
`pip install -r src/requirements.txt`

to get IPython and allow me to run >kedro ipython.  Correct?

**Comment by dwinski** [comment]
Otherwise, how did you have IPython in this environment?

**Comment by dwinski** [comment]
Ok, after recreating environment per your steps I just ran:
`pip install -r src/requirements.txt`

... and now it's working!  No idea what the problem was with the other environment but at least I can keep progressing on tutorial now.  Thanks for your help.  Will post again if I hit a wall with tutorial and can't solve it with what help is already available on interwebs.
